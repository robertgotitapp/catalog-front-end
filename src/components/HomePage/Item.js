import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from 'react-bootstrap/Card';

export class Item extends Component {
  render() {
    return (
      <Card border="dark">
        <Card.Header as="h4" name='name'>{this.props.item.name}</Card.Header>
        <Card.Body name='price'>
          Price: $
          {this.props.item.price}
        </Card.Body>
        <Card.Text name='description'>
          {this.props.item.description}
        </Card.Text>
      </Card>
    );
  }
}

function mapStateToProps({ items }, { id }) {
  return {
    item: items.items[id],
  };
}

export default connect(mapStateToProps)(Item);
