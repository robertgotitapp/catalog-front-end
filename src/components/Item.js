import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from 'react-bootstrap/Card';

export class Item extends Component {
  render() {
    return (
      <Card border="dark" className='item-card'>
        <Card.Header as="h4">{this.props.item.name}</Card.Header>
        <Card.Body>
          Price: $
          {this.props.item.price}
        </Card.Body>
        <Card.Text>
          {this.props.item.description}
        </Card.Text>
      </Card>
    );
  }
}

function mapStateToProps({ itemsReducer }, { id }) {
  return {
    item: itemsReducer.items[id],
  };
}

export default connect(mapStateToProps)(Item);
