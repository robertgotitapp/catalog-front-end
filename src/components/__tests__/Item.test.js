import React from 'react';
import { shallow } from 'enzyme';
import { Item } from '../Item';

describe('components/Item', () => {
  const props = {
    item: {
      category_id: 1,
      created: '2019-06-17T13:49:00+00:00',
      description: 'Item 4',
      id: 11,
      name: 'Item 4',
      price: 19,
      updated: '2019-06-17T13:49:00+00:00',
    },
  };
  const container = shallow(<Item {...props} />);

  it('should render correctly', () => {
    expect(container).toMatchSnapshot();
  });

  it('should contain exact number of item name', () => {
    expect(container.find('[name="name"]')).toHaveLength(1);
  });

  it('should contain exact number of item price', () => {
    expect(container.find('[name="price"]')).toHaveLength(1);
  });

  it('should contain exact number of item description', () => {
    expect(container.find('[name="description"]')).toHaveLength(1);
  });
});
