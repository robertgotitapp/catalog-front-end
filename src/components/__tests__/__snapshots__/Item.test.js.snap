// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Item should render correctly 1`] = `
<Card
  body={false}
  border="dark"
  className="item-card"
>
  <CardHeader
    as="h4"
    name="name"
  >
    Item 4
  </CardHeader>
  <CardBody
    name="price"
  >
    Price: $
    19
  </CardBody>
  <CardText
    name="description"
  >
    Item 4
  </CardText>
</Card>
`;
